Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+StateMachine.swift

```diff
@@ -81,41 +98,40 @@ extension HTTPConnectionPool {
         }
 
         mutating func executeRequest(_ request: Request) -> Action {
-            switch self.state {
-            case .http1(var http1StateMachine):
-                let action = http1StateMachine.executeRequest(request)
-                self.state = .http1(http1StateMachine)
-                return action
+            self.state.modify { http1 in
+                http1.executeRequest(request)
+            } http2: { http2 in
+                http2.executeRequest(request)
```

Comment:
I'm afraid this won't work with Swift 5.2:
https://github.com/apple/swift-evolution/blob/main/proposals/0279-multiple-trailing-closures.md